### PR TITLE
Fix template and test bugs from PR #4541

### PR DESCRIPTION
**Pull request by paulomorgado**
Sorry about that everyone.

@MaximeMorin-Devolutions, can you check the changes and propose changes for anything else you found?

### CODE REVIEW

**Comment by MaximeMorin-Devolutions** [comment]
@paulomorgado LGTM, thanks for the quick fixes!

**Review by MaximeMorin-Devolutions — APPROVED** [review]

File: src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.liquid [hunk 1]
```diff
@@ -35,11 +35,11 @@
 {% else -%}
     public {{ Class }}()
     {
-{% endif -%}
-{% if UseBaseUrl and GenerateBaseUrlProperty -%}
+{%     if UseBaseUrl and GenerateBaseUrlProperty -%}
         BaseUrl = "{{ BaseUrl }}";
+{%     endif -%}
```
Comment (ramonsmits) [inline src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.liquid:40]: Something seems of with BaseUrl, the property isn't assigned due to this when run with NSwagStudio 13.20.0.0.

Comment (paulomorgado) [reply src/NSwag.CodeGeneration.CSharp/Templates/Client.Class.liquid:40]: Are you replacing NSwagStudio 13.20.0.0 templates with this one?

Can you share a simple repro?

**Comment by RicoSuter** [comment]
Is this ready to merge and will fix the problem? What about the open comment?

**Comment by paulomorgado** [comment]
> Is this ready to merge and will fix the problem? What about the open comment?

As far as I could assert from the tests, yes!

I have more coming up, but I want to keep them bite sized.

**Pull request merged**